lib/src/components/NxTransferListHalf/__tests__/NxTransferListHalf.test.tsx

```diff
@@ -4,495 +4,537 @@
+      });
 
-    expect(item).toExist();
-    expect(item).toMatchSelector('div');
+  it('fires onMoveAll when the "Remove All"/ "Transfer All" button is clicked', async function() {
+    const user = userEvent.setup(),
+        onMoveAll = jest.fn(),
+        moveBtn = quickRender({ onMoveAll, showMoveAll: true }).getByRole('button', { name: /remove all/i });
 
-    expect(select).toExist();
-    expect(select).toMatchSelector('label');
-    expect(select).toHaveText('foo');
+    expect(onMoveAll).not.toHaveBeenCalled();
+    await user.click(moveBtn);
+    expect(onMoveAll).toHaveBeenCalled();
+  });
 
-    expect(icon).toExist();
-    expect(icon).toHaveProp('icon', faTimesCircle);
+  it('passes onMoveAll an array containing every id if no filterValue is set', async function() {
+    const dataItems = [{
+      id: '1',
+      displayName: 'foo'
+    }, {
+      id: '2',
+      displayName: 'Foo'
+    }, {
+      id: '3',
+      displayName: 'bar'
+    }, {
+      id: '4',
+      displayName: 'foo'
+    }, {
+      id: '5',
+      displayName: 'Foo'
+    }, {
+      id: '6',
+      displayName: 'bar'
+    }];
 
-    expect(checkbox).toExist();
-    expect(checkbox).toHaveProp('type', 'checkbox');
+    const user = userEvent.setup(),
+        onMoveAll = jest.fn(),
+        moveBtn = quickRender({ items: dataItems, onMoveAll, showMoveAll: true })
+            .getByRole('button', { name: /remove all/i });
 
-    const otherItem = list.find('.nx-transfer-list__item').at(1);
+    expect(onMoveAll).not.toHaveBeenCalled();
+    await user.click(moveBtn);
 
-    expect(otherItem).toHaveText('baz');
+    expect(onMoveAll).toHaveBeenCalledWith(['1', '2', '3', '4', '5', '6']);
   });
 
-  it('does not render move icon and checkbox input when onItemChange is not provided', function() {
-    const items = [{
-          id: 1,
-          displayName: 'foo'
-        }],
-        componentDisableItemMove = getMounted({ items }),
-        select = componentDisableItemMove
-            .find('.nx-transfer-list__item-list .nx-transfer-list__item .nx-transfer-list__select');
+  it('renders items into the .nx-transfer-list__item-list', function() {
+    const dataItems = [{
+      id: '1',
+      displayName: 'foo'
+    }, {
+      id: '2',
+      displayName: 'baz'
+    }];
 
-    expect(select.find(NxFontAwesomeIcon)).not.toExist();
-    expect(select.find('input.nx-transfer-list__checkbox')).not.toExist();
+    const { container } = quickRender({ items: dataItems }),
+        itemList = container.querySelector('.nx-transfer-list__item-list') as HTMLElement,
+        items = itemList.querySelectorAll('.nx-transfer-list__item');
+
+    expect(items.length).toBe(2);
+    expect(items[0]).toHaveTextContent('foo');
+    expect(items[1]).toHaveTextContent('baz');
   });
 
-  it('sets the item checked prop to isSelected', function() {
+  it('renders a move icon and checkbox input only when onItemChange is provided', function() {
```

Comment:
> Do we have visual tests that adequately check the conditional around the icon? I think that would be the better way to test it than here

We didn't have any visual tests that explicitly laid out when the icon was visible or not, so added visual tests here: [ac29087](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/ac290870d4a087274abd1a214082c01838320147). Since we can't test which icon is being used when `isSelected` is true or false, I added clarity of which icon is rendered when.